### PR TITLE
Fix api key loading if RubyGems host is development

### PR DESCRIPTION
**Pull request by jenshenny**
## What was the end-user or developer problem that led to this PR?

From: https://github.com/rubygems/rubygems/pull/6624#issuecomment-1546185377

The `gem signin` fails with a warning if the host is `localhost:3000`
```
RUBYGEMS_HOST=http://localhost:3000 ruby -Ilib bin/gem signin
Failed to load /Users/jennyshen/.local/share/gem/credentials because it doesn't contain valid YAML hash
Failed to load /Users/jennyshen/.local/share/gem/credentials because it doesn't contain valid YAML hash
Enter your http://localhost:3000 credentials.
Don't have an account yet? Create one at http://localhost:3000/sign_up
   Email:   test2@example.com
Password:   

API Key name [Jennys-MacBook-Pro-2.local-jennyshen-20230516143333]:   
Please select scopes you want to enable for the API key (y/n)
index_rubygems [yN]  
...
Signed in with API key: Jennys-MacBook-Pro-2.local-jennyshen-20230516143333.
Failed to load /Users/jennyshen/.local/share/gem/credentials because it doesn't contain valid YAML hash
Failed to load /Users/jennyshen/.local/share/gem/credentials because it doesn't contain valid YAML hash
```

The original PR checks for invalid yaml like `invalid: foo: bar` by checking if there is `:` in the key. It is possible to have `:` as a part of the key like `http://localhost:3000`.

## What is your fix for the problem, implemented in this PR?
Change the `includes?(":")` to `includes?(": ")` as that would prompt invalid yaml.

Furthermore, after the warning was fixed, keys with `http://localhost:3000` still aren't successfully loaded. This is because

When the api keys gets loaded, the localhost key has a trailing `/` causing  `Gem.configuration.api_keys.key?(host)` to fail.

```
[41, 50] in /Users/jennyshen/src/github.com/Shopify/rubygems/lib/rubygems/gemcutter_utilities.rb
   41:   ##
   42:   # The API key from the command options or from the user's configuration.
   43: 
   44:   def api_key
   45:     byebug
=> 46:     if ENV["GEM_HOST_API_KEY"]
   47:       ENV["GEM_HOST_API_KEY"]
   48:     elsif options[:key]
   49:       verify_api_key options[:key]
   50:     elsif Gem.configuration.api_keys.key?(host)
(byebug) Gem.configuration.api_keys
{"http://localhost:3000/"=>"rubygems_"}
(byebug) Gem.configuration.api_keys.key?(host)
false
```

The serialized key gets converted back [here](https://github.com/rubygems/rubygems/blob/5351e01b32cd8a84d27f0d2dff27a229905543e1/lib/rubygems/config_file.rb#L529)

  - If the key contains `__`, it gets replaced with `.` and removes the trailing forward slash.

In the case of `localhost:3000`, since there aren't any `__` because the original host doesn't contain any `.` , the forward slash never gets removed. I changed the elsif to remove any trailing forward slash.

cc: @hsbt 

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [x] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

**Review by ericherscovich — COMMENTED** [review]
Looks good thank you!

**Review by indirect — APPROVED** [review]
looks good to me, let's get a review from @hsbt or @deivid-rodriguez to double check 👍🏻

**Review by deivid-rodriguez — APPROVED** [review]
It looks good to me!

**Review by simi — APPROVED** [review]

**Comment by jenshenny** [comment]
Would anyone be able to merge this in? I don't have merge access for this repo 😄

**Comment by indirect** [comment]
done!

**Pull request merged**
